src/interface/swarm.cpp

```diff
@@ -437,23 +406,25 @@ void Swarm::Defrag() {
 
   auto &intVector_ = std::get<getType<int>()>(Vectors_);
   auto &realVector_ = std::get<getType<Real>()>(Vectors_);
-  SwarmVariablePack<Real> vreal;
-  SwarmVariablePack<int> vint;
-  PackIndexMap rmap;
-  PackIndexMap imap;
-  vreal = PackAllVariables<Real>(rmap);
-  vint = PackAllVariables<int>(imap);
+  PackIndexMap real_imap;
+  PackIndexMap int_imap;
+  auto vreal = PackAllVariables_<Real>(real_imap);
+  auto vint = PackAllVariables_<int>(int_imap);
   int real_vars_size = realVector_.size();
   int int_vars_size = intVector_.size();
+  auto real_map = real_imap.Map();
+  auto int_map = int_imap.Map();
+  const int realPackDim = vreal.GetDim(2);
+  const int intPackDim = vint.GetDim(2);
 
   pmb->par_for(
       "Swarm::DefragVariables", 0, max_active_index_, KOKKOS_LAMBDA(const int n) {
         if (from_to_indices(n) >= 0) {
-          for (int i = 0; i < real_vars_size; i++) {
-            vreal(i, from_to_indices(n)) = vreal(i, n);
+          for (int vidx = 0; vidx < realPackDim; vidx++) {
+            vreal(vidx, from_to_indices(n)) = vreal(vidx, n);
```

Comment:
This no longer renders in the diff, but standardizing the swarm packing cleaned up a lot of weirdness here (extra `ParArrayND`s to store the dimensions of each packed variable, basically).